cgi-bin/LJ/User.pm

```diff
@@ -3195,61 +3198,121 @@ sub thread_expand_all {
     return 0;
 }
 
-#get/set Sticky Entry parent ID for settings menu
-sub sticky_entry {
-    my ( $u, $input ) = @_;
+# get/set Sticky Entry parent ID for settings menu
+# Expects an array of entry URLs or ditemids as input
+# If used as a setter, returns 1 or undef
+# Otherwise, returns an array of entry objects
+sub sticky_entries {
+    my ( $u, $input_ref ) = @_;
+
+    # The user may have previously had an account type that allowed more stickes.
+    # we want to preserve a record of these additional stickes in case they once
+    # more upgrade their account.  This means we must first extract these
+    # if they exist.
+    my $sticky_entry_ids =  $u->prop( 'sticky_entry' );
+    my @entry_ids = split( /,/, $sticky_entry_ids );
+
+    my $max_sticky_count = $u->count_max_stickies;
+    my $max_sticky_count_from_zero = $max_sticky_count - 1;
+    my $entry_length = @entry_ids;
+
+    my @currently_unused_stickies = @entry_ids[$max_sticky_count..$entry_length];
 
-    if ( defined $input ) {
-        unless ( $input ) {
+    # Check we've been sent input and it isn't empty.  If so we need to alter the sticky entries stored.
+    if ( defined $input_ref ) {
+        my $remote = LJ::get_remote();
```

Comment:
This method is on the `LJ::User` object, which means it shouldn't be checking `$remote`. That should be done at the input -- i.e., the API/page/etc that is making the call.